app/src/main/java/com/example/tasky/agenda/data/remote/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,19 @@
+package com.example.tasky.agenda.data.remote
+
+import com.example.tasky.agenda.data.remote.util.mapToAgenda
+import com.example.tasky.agenda.domain.AgendaRepository
+import com.example.tasky.agenda.presentation.model.Agenda
+import com.example.tasky.core.data.Resource
+import com.example.tasky.core.data.remote.safeApiCall
+import javax.inject.Inject
+
+class AgendaRepositoryImpl @Inject constructor(
+    private val agendaService: AgendaService
+) : AgendaRepository {
+
+    override suspend fun getAgenda(): Resource<List<Agenda>> {
+        return safeApiCall {
+            agendaService.getAgenda(time = System.currentTimeMillis())
+        }.mapToAgenda()
```

Comment:
(Read my other comments before reading this one)

Since Tasky is an offline-first app. That means the local db is the Single Source of Truth. Meaning everything from the API will be stored in the DB before being displayed

That's why you'll convert from DTO to Entity. Which will cause the duplication of code from the other comment